.github/workflows/main.yml

```diff
@@ -63,3 +81,19 @@ jobs:
           git -c user.name='ci' -c user.email='ci' commit -m init
           git push -f -q https://git:${{ secrets.github_token }}@github.com/${{ github.repository }} HEAD:gh-pages
         if: github.event_name == 'push' && github.event.ref == 'refs/heads/master'
+
+  msrv:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-14, windows-latest]
+    steps:
+    - uses: actions/checkout@master
+    - name: Install Rust (rustup)
+      run: rustup toolchain install nightly --no-self-update --profile minimal
+      shell: bash
+
+    - uses: taiki-e/install-action@cargo-hack
+    - uses: Swatinem/rust-cache@v2
+
+    - run: cargo hack check --lib --rust-version --ignore-private
```

Comment:
If the regular test job is not testing with the old dependencies, this job should

Reply:
I can add that back, but that would mean our `dev-dependencies` would also need to be below the msrv

Reply:
Alternatively, your primary test jobs could test with your lockfile and then this just needs to verify things compile.

(your primary test jobs should be testing with your lockfile anyways because you should isolate external factors from failing a PR build)

Reply:
Updated tests and msrv to use `--locked`

Reply:
Also fixed the testing by removing `dev-dependencies` that are too old and upgrade libc to 0.2.72